i18n-helpers/src/bin/mdbook-xgettext.rs

```diff
@@ -77,7 +77,7 @@ fn create_catalog(ctx: &RenderContext) -> anyhow::Result<Catalog> {
         })?;
         last_idx += idx;
         let lineno = summary[..last_idx].lines().count();
-        let source = format!("{}:{}", summary_path.display(), lineno);
+        let source = format!("{}:{lineno}", summary_path.display());
```

Comment:
I'm personally a bit in doubt about the mixed inlining and I think that's why I didn't inline it already. One approach I've seen people use is to do

```rust
format!("{foo}:{bar}", foo = complex + expressions)
```

I don't see it often, though, so I don't know if we want to bother with it. In any case, I'm fine with the partial inlining as well as a good first step.

Reply:
@qwandor what do you think of the style above? I'm not 100% sure I like it mself :smile:

Reply:
Personally I prefer the original style where nothing is inline.

Reply:
Thx for all the feedback.  I found the inline **tends to be** more readable if most/all of the values are inlined, and also it kinda forces me to move complex computations to a separate line, which helps readability as well.  I guess this specific one (tiny format string just to join two long values with a colon) is indeed more readable without inlining.

Reply:
Yeah, I agree that it looks better in _some_ but not all cases :-) Having to create a local variable instead of using `summary_path.display()` directly is annoying for example. I often wish we could have just a _bit_ more complex expressions here — like in Python or JavaScript. However, that's a discussion for others to have :smile: